toolkit/scripts/sodiff/mariner-sodiff.sh

```diff
@@ -9,41 +9,81 @@ repo_file_path="$2"
 mariner_version="$3"
 sodiff_out_dir="$4"
 sodiff_log_file="${sodiff_out_dir}/sodiff.log"
+current_os=$(cat /etc/os-release | grep ^ID | cut -d'=' -f2)
 
 # Setup output dir
 mkdir -p "$sodiff_out_dir"
+# Empty the log file
+echo > "$sodiff_log_file"
 
-# Prepare mariner/ubuntu compatibility calls
+function makecache_with_common {
+    if [[ "$current_os" == mariner ]]; then
+        # Mariner uses DNF repoquery command
+        DNF_COMMAND=dnf
+        # Cache RPM metadata
+        >/dev/null dnf -c "$1" --releasever $mariner_version -y makecache
```

Comment:
Let's also mage `mariner_vesion` and argument. In general I'd avoid using global variables, since they tend to make it harder to track what is happening and what a specific function depends on.